spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/actions/BaseRewriteDataFilesSparkAction.java

```diff
@@ -356,6 +374,8 @@ private void validateAndInitOptions() {
         "Cannot use options %s, they are not supported by the action or the strategy %s",
         invalidKeys, strategy.name());
 
+    strategy = strategy.options(options());
```

Comment:
What is this doing?

Reply:
ah I see we are moving this into the validate function. That's ok

Reply:
Yeah, while calling the validate function from test I saw the strategy options were not being initialised. Hence, moved it here. This helps in getting the different bytes/files for the groups too.